consensus/propagation/have_wants.go

```diff
@@ -216,6 +216,8 @@ func (blockProp *Reactor) handleWants(peer p2p.ID, wants *proptypes.WantParts) {
 	// send the peer the partset header if they don't have the proposal.
 	// TODO get rid of this catchup case
 	if round < 0 {
+		// TODO do we still need this? since we're not covering the gaps,
+		// we only catchup in the case we didn't receive all the haves/parts for the previous height/round.
```

Comment:
same

Reply:
if we're not covering gaps we don't and 

> round < 0

is likely a bug iirc since the round is sometimes set to < 0

Reply:
f3c1f5c49b6922eac3e4e38ac3c8f7ee038c7552

Reply:
f3c1f5c49b6922eac3e4e38ac3c8f7ee038c7552